### PR TITLE
Clean up some build warnings

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Fix some build warnings (only 649 more to go)

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [x] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [x] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Core/Bloom.cs [hunk 1]
```diff
@@ -18,17 +18,20 @@ public Bloom()
             Bytes = new byte[ByteLength];
         }
 
-        public Bloom(Bloom[] blooms) : this()
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Core/Bloom.cs:21]: Needs null handling

File: src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.PayloadProduction.cs [hunk 2]
```diff
@@ -55,6 +55,7 @@ public async Task getPayloadV1_should_allow_asking_multiple_times_by_same_payloa
     }
 
     [Test]
+    [Obsolete]
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.PayloadProduction.cs:58]: why?

File: src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.PayloadProduction.cs [hunk 3]
```diff
@@ -269,6 +270,7 @@ public async Task getPayload_should_serialize_unknown_payload_response_properly(
 
     [Test]
     [Retry(3)]
+    [Obsolete]
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.PayloadProduction.cs:273]: why?

File: src/Nethermind/Nethermind.Synchronization.Test/Mocks/FirstFree.cs [hunk 4]
```diff
@@ -31,9 +31,9 @@ private FirstFree()
 
         public bool CanBeReplaced => false;
 
-        public PeerInfo Allocate(PeerInfo currentPeer, IEnumerable<PeerInfo> peers, INodeStatsManager nodeStatsManager, IBlockTree blockTree)
+        public PeerInfo Allocate(PeerInfo? currentPeer, IEnumerable<PeerInfo> peers, INodeStatsManager nodeStatsManager, IBlockTree blockTree)
         {
-            return peers.FirstOrDefault() ?? currentPeer;
+            return peers.FirstOrDefault() ?? currentPeer!;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Synchronization.Test/Mocks/FirstFree.cs:36]: Doesn't make much sense to me

**Review by LukaszRozmej — COMMENTED** [review]
I wouldn't `Obsolete` those tests, they are still valid for example in Gnosis scenario.

File: src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.PayloadProduction.cs (hunk 2 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.PayloadProduction.cs:58]: Build warning as calls a method marked as Obsolete

File: src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.PayloadProduction.cs (hunk 3 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Merge.Plugin.Test/EngineModuleTests.PayloadProduction.cs:273]: Build warning as calls a method marked as Obsolete

File: src/Nethermind/Nethermind.Synchronization.Test/Mocks/FirstFree.cs (hunk 4 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Synchronization.Test/Mocks/FirstFree.cs:36]: Either peers has an item or has a currentPeer?

**Comment by benaadams** [comment]
> I wouldn't `Obsolete` those tests, they are still valid for example in Gnosis scenario.

Is just build warnings so they can call `[Obsolete]` items e.g.

image

File: src/Nethermind/Nethermind.Core/Bloom.cs (hunk 1 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Core/Bloom.cs:21]: Added

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**